collector/telemetry_collector.go

```diff
@@ -745,3 +789,315 @@ func (t *TelemetryCollector) emitResetMetrics(ch chan<- prometheus.Metric, label
+			}
+			if metricsByPort[gpuID][portID] == nil {
+				metricsByPort[gpuID][portID] = make(map[string]interface{})
+			}
+			metricsByPort[gpuID][portID][metricName] = metricValue.MetricValue
+		} else {
+			// Parse numeric metric value
+			value, err := parseMetricValue(metricValue.MetricValue)
+			if err != nil {
+				t.logger.Debug("failed to parse port metric value",
+					slog.String("metric", metricName),
+					slog.String("value", metricValue.MetricValue),
+					slog.Any("error", err),
+				)
+				continue
+			}
+
+			// Initialize maps if needed
+			if metricsByPort[gpuID] == nil {
+				metricsByPort[gpuID] = make(map[string]map[string]interface{})
+			}
+			if metricsByPort[gpuID][portID] == nil {
+				metricsByPort[gpuID][portID] = make(map[string]interface{})
+			}
+			metricsByPort[gpuID][portID][metricName] = value
+		}
+	}
+
+	// Extract system ID from metric properties (all should be same system)
+	systemID := extractSystemIDFromReport(report)
+	systemName := systemMap[systemID]
+	if systemName == "" {
+		systemName = systemID
+	}
+
+	// Emit metrics for each port
+	for gpuID, ports := range metricsByPort {
+		for portID, metrics := range ports {
+			labels := []string{systemName, systemID, gpuID, portID}
+			t.emitPortMetrics(ch, labels, metrics)
+		}
+	}
+}
+
+// parsePortMetricProperty extracts GPU ID, port ID, and metric name from a port MetricProperty path
+// Example: /redfish/v1/Systems/HGX_Baseboard_0/Processors/GPU_SXM_0/Ports/NVLink_0/Metrics#/RXBytes
+// Returns: "GPU_SXM_0", "NVLink_0", "RXBytes"
+func parsePortMetricProperty(property string) (gpuID string, portID string, metricName string) {
+	// Split on '#' to separate resource path from property path
+	parts := strings.Split(property, "#/")
+	if len(parts) != 2 {
+		return "", "", ""
+	}
+
+	resourcePath := parts[0]
+	metricName = parts[1]
+
+	// Extract GPU ID from resource path: /Processors/GPU_SXM_X/
+	if idx := strings.Index(resourcePath, "/Processors/"); idx != -1 {
+		remainder := resourcePath[idx+len("/Processors/"):]
+		if endIdx := strings.Index(remainder, "/"); endIdx != -1 {
+			gpuID = remainder[:endIdx]
+		}
+	}
+
+	// Extract Port ID from resource path: /Ports/PORT_ID/
+	if idx := strings.Index(resourcePath, "/Ports/"); idx != -1 {
+		remainder := resourcePath[idx+len("/Ports/"):]
+		if endIdx := strings.Index(remainder, "/"); endIdx != -1 {
+			portID = remainder[:endIdx]
+		}
+	}
+
+	return gpuID, portID, metricName
+}
+
+// emitPortMetrics emits Prometheus metrics for a single port
+func (t *TelemetryCollector) emitPortMetrics(ch chan<- prometheus.Metric, labels []string, metrics map[string]interface{}) {
+	// Link speed
+	if val, ok := metrics["CurrentSpeedGbps"].(float64); ok {
+		ch <- prometheus.MustNewConstMetric(
+			t.metrics["telemetry_port_current_speed_gbps"].desc,
+			prometheus.GaugeValue,
+			val,
+			labels...,
+		)
+	}
+
+	// Standard networking metrics - cumulative counters
+	if val, ok := metrics["RXBytes"].(float64); ok {
```

Comment:
I think most of new metrics are actually integer instead of float because they are cumulative values as counters

Reply:
Oh, really. Didn't know that. TIL.
Thank you for looking into this